### PR TITLE
feat(transformation): libraries import extractor

### PR DESCRIPTION
**Pull request by dparveez-ruddr**
## Description of the change

API to extract names of imported libraries from JS or python(faas) code.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -2,6 +2,9 @@ name: codacy-coverage-reporter
 
 on: push
 
+env:
+  FAAS_BASE_IMG: rudderlabs/openfaas-flask:feat.pythonLibraries
+
```
Comment (Jayachand) [inline .github/workflows/test.yml:7]: remove before final push

File: src/util/libExtractor.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const { parserForImport } = require('./parser');
+
+// param 'validateImports' supported for python/pythonfaas.
+async function extractLibraries(code, validateImports, language = "javascript") {
+    return parserForImport(code, validateImports, language);
+}
+
+exports.extractLibraries = extractLibraries;
```
Comment (Jayachand) [inline src/util/libExtractor.js:8]: nit: add this in parser.js. New file is not necessary

File: src/util/parser.js [hunk 3]
```diff
@@ -21,4 +37,27 @@ function parserForImport(code) {
   return obj;
 }
 
-exports.parserForImport = parserForImport;
+async function parserForPythonImports(code, validateImports=true, additionalLibs=[]) {
+  const obj = {};
+
+  const payload = [{
+    message: {
+      messageId: "-",
```
Comment (Jayachand) [inline src/util/parser.js:45]: Assign randomId or constant id. `-` looks odd

File: src/versionedRouter.js [hunk 4]
```diff
@@ -594,6 +595,17 @@ if (startDestTransformer) {
   });
 
   if (functionsEnabled()) {
+    router.post('/extractLibs', async (ctx) => {
+      try {
+        const { code, validateImports = false, language = "javascript" } = ctx.request.body;
+        const obj = await extractLibraries(code, validateImports, language);
```
Comment (Jayachand) [inline src/versionedRouter.js:601]:
```suggestion
       if (!code) {
          throw new Error('Invalid request. code is missing')
       }
        const obj = await extractLibraries(code, validateImports, language);
```

**Review by Jayachand — APPROVED** [review]

**Review by lvrach — APPROVED** [review]

**Review by sivashanmukh — APPROVED** [review]
LGTM

**Pull request merged**